From b9d41b784657152a4c5683a9458dffab557f0689 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Fri, 29 Aug 2014 18:58:56 +0300
Subject: [PATCH] alsactl: don't let systemd unit restore the volume when
 asound.state is missing

This avoids an error on bootup

Filed as https://bugtrack.alsa-project.org/alsa-bug/view.php?id=5459

Upstream-Status: Pending

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
Signed-off-by: Cristian Iorga <cristian.iorga@intel.com>

Rebased on 1.1.0.

Signed-off-by: Tanu Kaskinen <tanuk@iki.fi>
---
 alsactl/Makefile.am             | 7 ++++---
 alsactl/alsa-restore.service.in | 1 +
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/alsactl/Makefile.am b/alsactl/Makefile.am
index 90fab9d..190525a 100644
--- a/alsactl/Makefile.am
+++ b/alsactl/Makefile.am
@@ -38,9 +38,10 @@ install-data-hook:
 endif
 
 edit = \
-	$(SED) -r -e 's,@sbindir\@,$(sbindir),g' \
-		  -e 's,@mydatadir\@,$(mydatadir),g' \
-		  -e 's,@daemonswitch\@,$(ALSACTL_DAEMONSWITCH),g' \
+	$(SED) -e 's,@localstatedir\@,$(localstatedir),g' \
+	       -e 's,@sbindir\@,$(sbindir),g' \
+	       -e 's,@mydatadir\@,$(mydatadir),g' \
+	       -e 's,@daemonswitch\@,$(ALSACTL_DAEMONSWITCH),g' \
 							< $< > $@ || rm $@
 
 alsa-state.service: alsa-state.service.in
diff --git a/alsactl/alsa-restore.service.in b/alsactl/alsa-restore.service.in
index 80fd5fd..6beb36d 100644
--- a/alsactl/alsa-restore.service.in
+++ b/alsactl/alsa-restore.service.in
@@ -7,6 +7,7 @@
 Description=Save/Restore Sound Card State
 ConditionPathExists=!@daemonswitch@
 ConditionPathExistsGlob=/dev/snd/control*
+ConditionPathExists=@localstatedir@/lib/alsa/asound.state
 
 [Service]
 Type=oneshot
-- 
2.14.2

